### DISCUSSION TITLE
Staying on older Web3Modal version 2

### DISCUSSION THREAD

**Discussion by hashcodedavidchin** [WalletConnect/walletconnect-monorepo#3301] [Developer Support] [answered]
For context, the dapp currently using Web3Modal version `2.0.0-rc.1`.  Updating to latest Web3Modal means requires a higher version of typescript thus requiring huge effort on updating other packages. I would like to minimal effort on resolving error below without having to upgrade. Thanks

`WebSocket connection to 'wss://w.bridge.walletconnect.org/?env=browser&host=domain.com&protocol=wc&version=1' failed: `

**Comment by glitch-txs** [comment]
you need to switch in the config the version to 2 (`version: 2`). For typescript issues you can add "SkipLibCheck: true" in your tsconfig.json file

**Comment by hashcodedavidchin** [comment, marked as answer]
Web3Modal v2.1.1 (last version to use typescript v4) seems to have issues with MetaMask mobile thus I had to update to latest v2.7.1. This however requires typescript v5 and even with `skipLibCheck` it just don't recognised it correctly. I have successfully implement uses CDN instead but in trade off no typings. Thanks for the hint @glitch-txs
